pages/accounts/[username]/index.tsx

```diff
@@ -6,16 +6,24 @@ import { UserInfoWithData } from "@components/UserInfoHeader/withData";
 import { SensorsGrid } from "@components/SensorsGrid";
 import { ParsedSensorType } from "@lib/hooks/usePublicSensors";
 import { useUserData } from "@lib/hooks/useUserData";
+import { getUserNames } from "@lib/requests/getUserNames";
 
 export const getServerSideProps: GetServerSideProps = async context => {
   try {
     const username = context.query.username;
     if (!username || Array.isArray(username)) return { notFound: true };
 
-    const accountData = await getAccountDataByUsername(username);
```

Comment:
I don't think we need to get all usernames and compare them on the client-side. Instead, we should lowercase the username in the Supabase SDK call inside the `getAccountDataByUsername` function and use supabase's case insensitive syntax to find the relevant user `.ilike('column', '%CaseInsensitive%')` instead of `.eq('column', 'Equal to')`. This way, the comparison happens on the server. What do you think?

Reply:
See [here](https://app.supabase.io/project/anucvyvtfhtroefhotbr/api/default?resource=user_profiles) ans search for "case".

Reply:
Good points, I've changed `getAccountDataByUsername` to use `.ilike("name", username)`. Note that I didn't add the `%`s as they would indicate that there could be more characters before or after the username. This way, we make sure that the comparison is strictly against the provided username - only case-insensitively.

---

A side note out of curiosity: To my understanding there wouldn't have been a client-side comparison of usernames with the previous solution since the call was inside `getServerSideProps`. It could have still led to long loading times with the surely expected 1 million users. But nothing client-side, or is there?

Reply:
> But nothing client-side, or is there?

True! However, the initial load of the page would be slow because of that. Your newer version is much better.